helmclient.go

```diff
@@ -273,6 +332,8 @@ func (c *Client) EnsureTillerInstalled(ctx context.Context) error {
 			t, err := c.newTunnel()
 			if IsTillerNotFound(err) {
 				return backoff.Permanent(microerror.Mask(err))
+			} else if upgradeTiller && IsTooManyResults(err) {
+				return microerror.Maskf(err, "too many tiller pods due to upgrade")
```

Comment:
This is to make it clear in the logs that an upgrade is happening.

Reply:
So we would check for the `TooManyResults` error when we call `EnsureTillerInstalled`?
I would vouch for no error at all and waiting instead. That would match the behaviour described in the godoc of this function IMO.

Reply:
I was over thinking the logging I'll remove. 

This return is from a closure that is called by the backoff. So we need to return an error and wait until we are sure tiller is ready. I think this matches the godoc.